File: P1S2_Ruby/src/aula_practicas.rb
require_relative 'aula'

class AulaPracticas < Aula
  def initialize(nombre)
    super
    @ordenadores = []
    puts "Creada aula #{@nombre} de prácticas"
  end

  def add_ordenador(ordenador)
    @ordenadores << ordenador
  end

  def initialize_copy(orig)
    super
    @ordenadores = []
  end

  def to_s
    s = "AulaPracticas{nombre=#{@nombre}, ordenadores=["
    @ordenadores.each { |o| s += "#{o}, " }

    s += ']'
  end
end

File: P1S2/P1S2Java/src/AulaPracticas.java
import java.util.ArrayList;

public class AulaPracticas extends Aula {

    private final ArrayList<Ordenador> ordenadores;

    public AulaPracticas(String nombre) {
        super(nombre);
        ordenadores = new ArrayList<>();
        System.out.println("prácticas.");
    }

    public void addOrdenador(Ordenador ordenador) {
        ordenadores.add(ordenador);
    }

    @Override
    public String toString() {
        return "AulaPracticas{" +
                "nombre='" + nombre + '\'' +
                ", ordenadores=" + ordenadores +
                '}';
    }
}
